Kernel/Interrupts/InterruptManagement.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <AK/FixedArray.h>
+#include <AK/Function.h>
+#include <AK/RefCounted.h>
+#include <AK/RefPtr.h>
+#include <AK/Types.h>
+#include <Kernel/ACPI/Definitions.h>
+#include <Kernel/Interrupts/GenericInterruptHandler.h>
+#include <Kernel/Interrupts/IOAPIC.h>
+#include <Kernel/Interrupts/IRQController.h>
+namespace Kernel {
+class InterruptManagement {
+public:
+    static InterruptManagement& the();
+    static void initialize();
+
+    static void switch_to_pic_mode();
+    static void switch_to_ioapic_mode();
+
+    virtual void enable(u8 interrupt_vector);
+    virtual void disable(u8 interrupt_vector);
+    virtual void eoi(u8 interrupt_vector);
+
+    //virtual void register_interrupt_handler(GenericInterruptHandler& handler, u8 interrupt_vector);
+
+    void enumerate_interrupt_handlers(Function<void(GenericInterruptHandler&)>);
+
+    IRQController& get_interrupt_controller(int index);
+
+protected:
+    InterruptManagement(bool create_default_controller);
+    FixedArray<IRQController*>& m_interrupt_controllers;
```

Comment:
Instead of allocating member objects like this, using `new`, just declare them as regular members:

```suggestion
    FixedArray<IRQController*> { 1 };
```
```
```suggestion
    FixedArray<IRQController*>& m_interrupt_controllers;
```
```suggestion
    FixedArray<IRQController*>& m_interrupt_controllers;
```

Reply:
I fixed that, thanks! :)